enhancements/ingress/admin-configured-ca-bundle.md

```diff
@@ -0,0 +1,240 @@
+- As an OpenShift cluster administrator, I want to configure admin CA certificate bundle which
+  should be used for verifying service's certificate by the OpenShift router, instead of
+  having to configure the same as destinationCACertificate in each of the created route object.
+- As an OpenShift developer, I want to create a CA trust bundle for Ingress operator usage,
+  which includes the OpenShift service CA and the admin provided CA bundle as the default CA
+  for verifying the service's certificate when a destination CA certificate is not configured
+  for a route.
+- As an OpenShift developer, I want to update the OpenShift router certificate cache whenever
+  the OpenShift service CA certificate or the admin configured CA certificate bundle is updated
+  or deleted.
+
+### Goals
+
+- An OpenShift administrator can configure a CA certificate bundle and expect that to be used
+  for verifying the service's certificate by the OpenShift router without defining the
+  destinationCACertificate in the route object created for the service.
+- Openshift Router should be updated seamlessly when the OpenShift service CA certificate or
+  the administrator configured CA certificate bundles are modified.
+
+### Non-Goals
+
+- OpenShift administrator defined CA certificate bundle will be used only for the traffic
+  through OpenShift router and for the routes created with re-encrypt termination type, any
+  other scenario is outside the scope of this proposal.
+
+## Proposal
+
+OpenShift Ingress Operator to create a CA trust bundle which will contain the default OpenShift
+service CA certificate and the custom user-supplied CA certificates bundle and makes it available
+to the OpenShift Router as the default destination CA bundle file to be used for verifying the
+services' certificate. And also updates the CA bundle file whenever either of the certificates
+change, which should be watched by the OpenShift Router and update the same internally.
+
+### Workflow Description
+
+- An OpenShift administrator configures the list of trusted CA certificates as a ConfigMap.
+- OpenShift Ingress Operator creates a ConfigMap with the admin configured CA certificates and the
+  OpenShift service CA certificate stored in the respective ConfigMaps.
+- OpenShift Ingress Operator creates a watcher and reconciles whenever changes are observed
+  in the aforementioned ConfigMap objects.
+- Openshift Router uses the operator created ConfigMap as the default destination CA certificate
+  to verify services' certificate, when a route with re-encrypt termination type is created
+  without destinationCA configured.
+- OpenShift router watches and reloads whenever CA certificate bundle changes.
+
+![alt Workflow diagram](admin-configured-ca-bundle.png)
+
+#### Variation [optional]
+None
+
+### API Extensions
+None
+
+### Implementation Details/Notes/Constraints [optional]
+
+An OpenShift administrator will have to create a ConfigMap with the fixed name `admin-ca-bundle`
```

Comment:
I thought to use the generic name with the intention, ca-bundle usage could be extended for any other similar use-cases as well.

Reply:
Hm, so it could be more general in the future, like a custom service CA?  @deads2k, I wonder whether you have an opinion on this.

Reply:
On further thought, would it make sense instead to define a label, and change cluster-ingress-operator to incorporate all CA certificates from all ConfigMaps with that label into the router's trust bundle?  

The reason I ask is because I was thinking about how something like cert-manager could use this feature, and it occurred to me that it would be much easier for independent simultaneous actors (humans or controllers) to use this feature to inject their respective certificates into the router's trust bundle if you used a well known label (allowing multiple ConfigMaps) instead of a well known name (requiring everything to be put into a single ConfigMap).